crates/ra_ide_api/src/call_info.rs

```diff
@@ -462,4 +476,77 @@ fn main() {
         assert_eq!(info.active_parameter, Some(1));
         assert_eq!(info.label(), "fn bar(&self, _: u32)");
     }
+
+    #[test]
+    fn works_for_tuple_structs() {
+        let info = call_info(
+            r#"
+/// A cool tuple struct
+struct TS(u32, i32);
+fn main() {
+    let s = TS(0, <|>);
+}"#,
+        );
+
+        assert_eq!(info.label(), "struct TS(u32, i32) -> TS");
+        assert_eq!(info.doc().map(|it| it.into()), Some("A cool tuple struct".to_string()));
+        assert_eq!(info.active_parameter, Some(1));
+    }
+
+    #[test]
+    #[should_panic]
```

Comment:
Hm, I've noticed that these should_panic's are still there. I guess that means that something's not entirely fixed yet?

Reply:
I think it's because `call_info` unconditionally returns `CallInfo`. If we change it to `Result<CallInfo>` we can probably get rid of the `should_panic`.

Reply:
Actually Option might fit better